modules/apps/client-extension/client-extension-service/src/main/java/com/liferay/client/extension/internal/service/taglib/ClientExtensionTopHeadDynamicInclude.java

```diff
@@ -61,13 +68,40 @@ public void include(
 				continue;
 			}
 
-			printWriter.print("<link data-senna-track=\"temporary\" href=\"");
-			printWriter.print(globalCSSCET.getURL());
-			printWriter.print(StringPool.QUOTE);
-			printWriter.print(
-				ContentSecurityPolicyNonceProviderUtil.getNonceAttribute(
-					httpServletRequest));
-			printWriter.print(" rel=\"stylesheet\" type=\"text/css\" />");
+			_writeStyleSheet(
+				httpServletRequest, printWriter, globalCSSCET.getURL());
+		}
+
+		if (!FeatureFlagManagerUtil.isEnabled(
+				themeDisplay.getCompanyId(), "LPD-34650")) {
+
+			return;
+		}
+
+		try {
+			List<CET> cets = _cetManager.getCETs(
+				themeDisplay.getCompanyId(), null,
+				ClientExtensionEntryConstants.TYPE_GLOBAL_CSS,
+				Pagination.of(QueryUtil.ALL_POS, QueryUtil.ALL_POS), null);
+
+			for (CET cet : cets) {
+				GlobalCSSCET globalCSSCET = (GlobalCSSCET)cet;
+
+				if (!StringUtil.equalsIgnoreCase(
+						globalCSSCET.getScope(), "company")) {
+
+					continue;
+				}
+
+				_writeStyleSheet(
+					httpServletRequest, printWriter, globalCSSCET.getURL());
+			}
+		}
+		catch (Exception exception) {
+			_log.error(
+				"Unable to inject global CSS client extensions for company " +
```

Comment:
We should just refer to this as "CSS client extension". "global" should be removed.